book.tex

```diff
@@ -1544,20 +1544,26 @@ \section{Flatten Expressions}
 from a function using the \key{values} form and you can receive
 multiple things from a function call using the \key{define-values}
 form. If you are not familiar with these constructs, the Racket
-documentation will be of help. Also, the \key{map3} function
+documentation will be of help.
+Also, the \key{map3} function
 (Appendix~\ref{appendix:utilities}) is useful for applying a function
 to each element of a list, in the case where the function returns
 three values. The result of \key{map3} is three lists.
 
 The clause of \key{flatten} for the \key{program} node needs to
 recursively flatten the body of the program and the newly flattened
-expression should be placed in a \key{return} statement.  The
-\key{flatten} pass should also compute the list of variables used in
-the program.  I recommend traversing the statements in the body of the
-program (after it has been flattened) and collect all variables that
-appear on the left-hand-side of an assignment. Note that each variable
-should only occur once in the list of variables that you place in the
-\key{program} form.
+expression should be placed in a \key{return} statement. Remember that
+the variable list in the \key{program} node should contain no duplicates.
+%% The
+%% \key{flatten} pass should also compute the list of variables used in
+%% the program.
+%% I recommend traversing the statements in the body of the
+%% program (after it has been flattened) and collect all variables that
+%% appear on the left-hand-side of an assignment.
+%% Note that each variable
+%% should only occur once in the list of variables that you place in the
+%% \key{program} form.
```

Comment:
The commented out prose recommended traversing the statement after it has been flattened to collect variables, but in the previous paragraph the book recommended accumulating those variables *during* flattening. Either way will work, but we have to pick one---this change makes the book consistent with the course compiler.